### PR TITLE
refactor(pg): add tests for CLI args

### PR DESCRIPTION
**Pull request by sam-goldman**
This PR adds coverage for the propose and deploy CLI command arguments. It involves a pretty large refactor to the CLI logic to make it possible to use dependency injection in these tests. Although a lot of code was moved around, the structure of the CLI commands remains the same.

This PR introduces [sinon](https://github.com/sinonjs/sinon), a popular JS mocking library, to remove boilerplate code from our tests that use mocks. With sinon, we can check that the arguments passed into functions like `propose` match what we expect. It removes the need to define mock functions like this in our tests:
```ts
let expectedScriptPath = ''
const mockPropose = (
  scriptPath: string,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  confirm: string,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  targetContract: string,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  silent: boolean,
  ...
) => {
  expectedScriptPath = scriptPath
}

expect(expectedScriptPath).equals(actualScriptPath)
```

### CODE REVIEW

**Comment by sam-goldman** [comment]
@RPate97 not necessary to review this. if you want to take a quick look, the most important part is that we have sufficient coverage for the CLI commands, which are all in [`cli.spec.ts`](https://github.com/sphinx-labs/sphinx/pull/1306/files#diff-e47308c4f4423a6b3f6ef13835d3194df936fe9dd73bc64696e885ea044ace55). Also, lmk what you think about using sinon

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+      expect(deploySpy.called).to.be.false
+      expect(consoleErrorSpy.called).to.be.true
+      expect(consoleErrorSpy.firstCall.args[0]).to.include(
+        'Missing required argument: network'
+      )
+    })
+
+    it('--network ethereum', () => {
+      const network = 'testnet'
+      const args = ['deploy', scriptPath, '--network', network]
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:222]:
```suggestion
    it('--network ethereum', () => {
      const network = 'ethereum'
      const args = ['deploy', scriptPath, '--network', network]
```

Minor suggestion, there are also a number of other test cases here with similar minor errors. Not a huge deal, but it is kind of confusing for someone trying to understand the tests.

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        verify: false,
+        targetContract: undefined,
+        forceRecompile: true,
+      }
+
+      expect(deploySpy.calledWithMatch(expectedParams)).to.be.true
+    })
+
+    it('--network ethereum --confirm', () => {
+      const args = ['deploy', scriptPath, '--network', 'testnet', '--confirm']
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:243]:
```suggestion
    it('--network ethereum --confirm', () => {
      const args = ['deploy', scriptPath, '--network', 'ethereum', '--confirm']
```

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+    it('--network ethereum --target-contract MyContract', () => {
+      const args = [
+        'deploy',
+        scriptPath,
+        '--network',
+        'testnet',
+        '--target-contract',
+        'MyContract',
+      ]
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:271]:
```suggestion
    it('--network ethereum --target-contract MyContract', () => {
      const args = [
        'deploy',
        scriptPath,
        '--network',
        'ethereum',
        '--target-contract',
        'MyContract',
      ]
```

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+
+  describe('deploy', () => {
+    let deploySpy: sinon.SinonStub
+
+    beforeEach(() => {
+      deploySpy = sinon.stub(sphinxContext, 'deploy')
+    })
+
+    it('fails if no script path is included', async () => {
+      const args = ['deploy', '--network', 'testnet']
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:197]:
```suggestion
    it('fails if no script path is included', async () => {
      const args = ['deploy', '--network', 'ethereum']
```

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+        'MyContract',
+      ]
+
+      makeCLI(args, sphinxContext)
+
+      expect(deploySpy.called).to.be.true
+
+      const expectedParams = {
+        scriptPath,
+        network: 'testnet',
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:279]:
```suggestion
        network: 'ethereum',
```

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    it('--network ethereum --verify', () => {
+      const args = ['deploy', scriptPath, '--network', 'testnet', '--verify']
+
+      makeCLI(args, sphinxContext)
+
+      expect(deploySpy.called).to.be.true
+
+      const expectedParams = {
+        scriptPath,
+        network: 'testnet',
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:300]:
```suggestion
      const args = ['deploy', scriptPath, '--network', 'ethereum', '--verify']

      makeCLI(args, sphinxContext)

      expect(deploySpy.called).to.be.true

      const expectedParams = {
        scriptPath,
        network: 'ethereum',
```

File: packages/plugins/test/mocha/cli/cli.spec.ts [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+      const args = ['deploy', scriptPath, '--network', 'testnet', '--silent']
+
+      makeCLI(args, sphinxContext)
+
+      expect(deploySpy.called).to.be.true
+
+      const expectedParams = {
+        scriptPath,
+        network: 'testnet',
+        skipPreview: false,
```
Comment (RPate97) [inline packages/plugins/test/mocha/cli/cli.spec.ts:322]:
```suggestion
    it('--network ethereum --silent', () => {
      const args = ['deploy', scriptPath, '--network', 'ethereum', '--silent']

      makeCLI(args, sphinxContext)

      expect(deploySpy.called).to.be.true

      const expectedParams = {
        scriptPath,
        network: 'ethereum',
        skipPreview: false,
```

**Review by RPate97 — APPROVED** [review]
Nice job, good refactor and good test coverage. I just left some minor suggestions. I am more than happy to use sinon.

File: packages/plugins/test/mocha/cli/cli.spec.ts (hunk 1 above)
Comment (sam-goldman) [reply packages/plugins/test/mocha/cli/cli.spec.ts:222]: Nice catch

**Pull request merged**